python.md

```diff
@@ -0,0 +1,1376 @@
+
+```python
+import os
+import rpm
+
+def read_rpm_header(ts, filename):
+    fd = os.open(filename, os.O_RDONLY)
+    h = None
+    try:
+        h = ts.hdrFromFdno(fd)
+    except rpm.error as ex:
+        error = str(ex)
+        if error == "public key not available":
+            print(error)
+        if error == "public key not trusted":
+            print(error)
+        if error == "error reading package header":
+            print(error)
+    finally:
+        os.close(fd)
+    return h
+
+ts = rpm.TransactionSet()
+h = read_rpm_header(ts, "n-v-r.rpm")
+```
+
+You can decide in your code whether the exceptions should stop processing or
+not.
+
+
+## Setting the verification flags
+
+Starting with rpm 4.1, package files are verified automatically, which can cause
+problems, especially if you are working with older packages, or packages without
+proper digital signatures.
+
+In most cases, the automatic verification is an advantage, since you can have
+greater confidence in the package files. However, you can call `setVSFlags` on a
+transaction set to change the default behavior.
+
+```python
+ts.setVSFlags(flags)
+```
+
+For example, if you have problems with old packages that do not have proper
+signatures, you can use code like the following to ignore such checks:
+
+```python
+# Set to not verify DSA signatures.
+ts.setVSFlags(rpm.RPMVSF_NODSA)
+```
+
+The following table lists the flags you can pass to `setVSFlags` on a
+transaction set. These flags are bitmasks. You can "or" them together for more
+than one setting. You must do a binary "or". Do not use the Python `or` keyword,
+use `|` instead, for a binary "or" operation.
+
+| Flag                       | Meaning                                                          |
+|----------------------------|------------------------------------------------------------------|
+| `rpm.RPMVSF_NEEDPAYLOAD`   | Leave the file handle positions at the beginning of the payload. |
+| `rpm.RPMVSF_NOHDRCHK`      | Don’t check the RPM database header.                             |
+| `rpm.RPMVSF_ NODSA`        | Don’t check the header and payload DSA signatures.               |
+| `rpm.RPMVSF_ NODSAHEADER`  | Don’t check the header DSA signature.                            |
+| `rpm.RPMVSF_ NOMD5`        | Don’t check the header and payload MD5 digests.                  |
+| `rpm.RPMVSF_ NORSA`        | Don’t check the header and payload RSA signatures.               |
+| `rpm.RPMVSF_ NOSHA1HEADER` | Don’t check the header SHA1 digest.                              |
+| `rpm._RPMVSF_NODIGESTS`    | Convenience to not check digests.                                |
+| `rpm._RPMVSF_NOSIGNATURES` | Convenience to not check signatures.                             |
+
+To turn off all checks, you can pass `–1` to `setVSFlags`:
+
+```python
+ts.setVSFlags(-1)
+```
+
+
+## Dependency Comparisons
+
+FIXME code examples in this chapter are outdated
```

Comment:
The code uses `file_h.dsOfHeader()`, I don't know what should be done instead. I know a syntax like `rpm.ds(h, rpm.RPMTAG_PROVIDENAME)` but it takes an argument, the original example doesn't. I was thinking of `rpm.ds(h, rpm.RPMTAG_EVR)` but then `.EVR()` on it returns nothing. So I don't know.